batch.go

```diff
@@ -122,41 +121,37 @@ func (b *Batch) createPushRequest() (*logproto.PushRequest, int) {
 	return &req, entriesCount
 }
 
-// generateEntries creates a list of log entries
-func generateEntries(ctx context.Context, tenantID string, pool LabelPool, numStreams, minBatchSize, maxBatchSize int) []Entry {
+// generateEntries creates a batch
+func generateEntries(ctx context.Context, tenantID string, pool LabelPool, numStreams, minBatchSize, maxBatchSize int) *Batch {
+	batch := &Batch{
+		Streams:   map[string]*logproto.Stream{},
+		CreatedAt: time.Now(),
+	}
 	state := lib.GetState(ctx)
 
 	hostname, err := os.Hostname()
 	if err != nil {
 		hostname = "localhost"
 	}
 
-	entries := make([]Entry, 0)
-	var labels model.LabelSet
 	maxSizePerStream := (minBatchSize + rand.Intn(maxBatchSize-minBatchSize)) / numStreams
-	batchSize := 0
 	lines := 0
 
 	for i := 0; i < numStreams; i++ {
-		labels = labelsFromPool(pool)
+		labels := labelsFromPool(pool)
 		labels[model.InstanceLabel] = model.LabelValue(fmt.Sprintf("vu%d.%s", state.VUID, hostname))
-
-		streamSize := 0
-		for streamSize < maxSizePerStream {
-			line := flog.NewLog(string(labels[model.LabelName("format")]), time.Now())
-			entry := Entry{
-				Entry: logproto.Entry{
-					Timestamp: time.Now(),
-					Line:      line,
-				},
-				TenantID: tenantID,
-				Labels:   labels,
-			}
-			entries = append(entries, entry)
-			streamSize += len(line)
-			lines++
+		stream := &logproto.Stream{Labels: labels.String()}
+		batch.Streams[stream.Labels] = stream
+
+		var line string
+		for ; batch.Bytes < maxSizePerStream; batch.Bytes += len(line) {
+			line = flog.NewLog(string(labels[model.LabelName("format")]), time.Now())
+			stream.Entries = append(stream.Entries, logproto.Entry{
+				Timestamp: time.Now(),
```

Comment:
These two calls to time.Now() are currently the *biggest* CPU user if you just run the `examples/simple.js` - we can make it into one in the begging of the `for` loop, but I guess you want to have the time in the line?